rust/agama-lib/src/product/client.rs

```diff
@@ -146,6 +146,28 @@ impl<'a> ProductClient<'a> {
         Ok(addons)
     }
 
+    // details of available addons
+    pub async fn available_addons(&self) -> Result<Vec<AddonProperties>, ServiceError> {
+        let addons: Vec<AddonProperties> = self
+            .registration_proxy
+            .available_addons()
+            .await?
+            .into_iter()
+            .map(|hash| AddonProperties {
```

Comment:
It would be a good idea to move this code to a separate function. But, for now, it is good enough. Thanks!